src/sentry/api/endpoints/organization_repository_commits.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+
+from sentry.api.base import DocSection
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.exceptions import ResourceNotFOund
+from sentry.api.serializers import serialize
+from sentry.models import Commit, Repository
+
+
+class OrganizationRepositoryCommitsEndpoint(OrganizationEndpoint):
+    doc_section = DocSection.ORGANIZATIONS
+
+    def get(self, request, organization, repo_id):
+        """
+        List a Repositories's Commits
+        `````````````````````````````
+
+        Return a list of version control repositories bound to a organization.
+
+        :pparam string organization_slug: the organization short name
+        :pparam string repo_id: the repository ID
+        :auth: required
+        """
+        try:
+            repo = Repository.objects.get(
+                id=repo_id,
+                organization_id=organization.id,
+            )
+        except Repository.DoesNotExist:
+            raise ResourceNotFOund
+
+        queryset = Commit.objects.filter(
+            repository_id=repo.id,
```

Comment:
I _think_ this will leverage the `repository_id, key` index here. Not a blocker now, but will want to double check before we forget.